api/api.go

```diff
@@ -770,13 +802,27 @@ func (a *API) service(w http.ResponseWriter, r *http.Request) {
 			agpList = list
 		}
 
+		// save ms def in local db
+		if service.SensorVersion != nil {
+			if err := persistence.SaveOrUpdateMicroserviceDef(a.db, msdef); err != nil { // save to db
+				glog.Errorf("Error saving microservice definition %v into db: %v", *msdef, err)
+				w.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+		}
+
 		// Generate a policy based on all the attributes and the service definition
-		if genErr := policy.GeneratePolicy(a.Messages(), *service.SensorName, policyArch, &props, haPartner, meterPolicy, counterPartyProperties, *agpList, a.Config.Edge.PolicyPath); genErr != nil {
+		if genErr := policy.GeneratePolicy(a.Messages(), *service.SensorUrl, *service.SensorName, service.SensorVersion, policyArch, &props, haPartner, meterPolicy, counterPartyProperties, *agpList, a.Config.Edge.PolicyPath); genErr != nil {
 			glog.Errorf("Error: %v", genErr)
 			w.WriteHeader(http.StatusInternalServerError)
 			return
 		}
 
+		// Fire event to load the microservice container
+		if service.SensorVersion != nil {
+			a.Messages() <- events.NewStartMicroserviceMessage(events.START_MICROSERVICE, msdef.SpecRef, msdef.Version, msdef.Arch)
```

Comment:
That is also okay. But since the start of microservice function/event might be used in other places too, the caller would have to call persistence to retrieve the microservice key and the start microservice function would have to call the persistence to retrieve the microservice again. This will have 2 hits to the db.